snow/networking/handler/handler.go

```diff
@@ -169,6 +170,10 @@ func (h *handler) SetEngineManager(engineManager *EngineManager) {
 	h.engineManager = engineManager
 }
 
+func (h *handler) GetEngineManager() *EngineManager {
```

Comment:
I think we don't need to hold the `ctx.Lock` on `Set` since we only call it when setting up the chain, but do we need to hold the lock here?

Reply:
We only modify the value when calling `SetEngineManager` - so this var isn't written in other cases... so it should be fine not to lock here.